main_test.go

```diff
@@ -0,0 +1,65 @@
+package main
+
+import (
+	"bytes"
+	"flag"
+	"net"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestUsage(t *testing.T) {
+	w := bytes.NewBuffer(nil)
+	flag.CommandLine.SetOutput(w)
+	usage()
+	opts := `[-p <port>] [-a] [<dir>]`
+	desc := `Simple HTTP server, serving files from given directory.`
+	require.Contains(t, w.String(), opts)
+	require.Contains(t, w.String(), desc)
+}
+
+func TestDir(t *testing.T) {
+	require.Equal(t, ".", dir(nil))
+	require.Equal(t, ".", dir([]string{}))
```

Comment:
I feel this is redundant (or the previous), since the nil and empty slice are defined to be the same (and almost are).

Reply:
ok, removed.